syntax = "proto3";
package report;

service ReportService {
  // Формирование отчёта
  rpc GenerateReport(ReportRequest) returns (ReportResponse);
  // Экспорт отчёта
  rpc GetReport(GetReportRequest) returns (ReportData);
}

message ReportRequest {
  string user_id = 1;
  int64 from_ts = 2;
  int64 to_ts = 3;
}
message ReportResponse {
  string report_id = 1;
  int64 total_income = 2;
  int64 total_expense = 3;
  map<string,double> by_category = 4;
}

message GetReportRequest {
  string report_id = 1;
}
message ReportData {
  string report_id = 1;
  string user_id = 2;
  int64 generated_at = 3;
  double total_income = 4;
  double total_expense = 5;
  map<string,double> by_category = 6;
  bytes payload = 7;
}
